### PR TITLE
WIP: Add pvc-cleaner to pipelines-service.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### Referenced issue:

https://issues.redhat.com/browse/PLNSRVCE-374

### Demo

https://youtu.be/WHGuVv1noz8

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: gitops/pvc-cleaner/pvc-cleaner.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  name: pvc-cleaner
+  namespace: openshift-gitops
+
+spec:
+  destination:
+    namespace: pvc-cleaner
+    server: https://kubernetes.default.svc
+  source:
+    path: gitops/pvc-cleaner/pvc-cleaner-app
+    repoURL: https://github.com/AndrienkoAleksandr/pipelines-service.git
```
Comment (Roming22) [inline gitops/pvc-cleaner/pvc-cleaner.yaml:13]: You should not reference your personal repository.

File: gitops/pvc-cleaner/pvc-cleaner-app/argo-pvc-cleaner-role.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: argocd-manager-role
+  namespace: pvc-cleaner
+rules:
```
Comment (Roming22) [inline gitops/pvc-cleaner/pvc-cleaner-app/argo-pvc-cleaner-role.yaml:6]: If the goal is to remove PVCs, those rules seems a bit to permissive.

File: gitops/argocd/argo-apps/pvc-cleaner.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: pvc-cleaner
+  namespace: openshift-gitops
+
+spec:
+  destination:
+    namespace: pvc-cleaner
```
Comment (ramessesii2) [inline gitops/argocd/argo-apps/pvc-cleaner.yaml:9]: Hey @AndrienkoAleksandr, I'm not sure about installing `pvc-cleaner` application in `pvc-cleaner` namespace. AFAIK, gitops have limited accessibility to other namespaces as per current design. In addition, following the current pattern, Argo CD applications are being installed in `openshift-gitops` ns. I would ask for @bnallapeta  's opinion on this.

**Review by ramessesii2 — COMMENTED** [review]
This looks good to me. I've just one comment though. Thanks.

File: gitops/pvc-cleaner/pvc-cleaner-app/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- namespace.yaml
+- argo-pvc-cleaner-role.yaml
+- argo-pvc-cleaner-rolebinding.yaml
+- https://github.com/redhat-appstudio/pvc-cleaner//infra-deployments?ref=96cb6db1991efa309334496fa0fd4494e9e75b23
```
[the remark this answers is not in the document]
Comment (AndrienkoAleksandr) [reply gitops/pvc-cleaner/pvc-cleaner-app/kustomization.yaml]: I wanted split folders from repo path, but ok I changed and used single slash.

File: gitops/argocd/argo-apps/pvc-cleaner.yaml (hunk 3 above)
Comment (AndrienkoAleksandr) [reply gitops/argocd/argo-apps/pvc-cleaner.yaml:9]: I guess to have wide access to the another namespaces for some cases we have permission workaronds https://github.com/openshift-pipelines/pipelines-service/tree/main/exploration/argocd-external/gitops/sre/environment/compute/base/argocd-rbac. For example https://github.com/openshift-pipelines/pipelines-service/blob/main/exploration/argocd-external/gitops/sre/environment/compute/base/argocd-rbac/argo-syncer-role.yaml and https://github.com/openshift-pipelines/pipelines-service/blob/main/exploration/argocd-external/gitops/sre/environment/compute/base/argocd-rbac/argo-tekton-role.yaml ... But I agree, that's interesting to get feedback from @bnallapeta

Comment (bnallapeta) [reply gitops/argocd/argo-apps/pvc-cleaner.yaml:9]: @AndrienkoAleksandr The link you shared is in the exploration folder for now. We wish to use this when we get to a state where one single Argo CD server is managing multiple applications from multiple clusters. 
But as of today, we are installing Argo CD server on each physical cluster and thus using the openshift-gitops namespace. Please change the namespace for the pvc-cleaner accordingly.

Comment (AndrienkoAleksandr) [reply gitops/argocd/argo-apps/pvc-cleaner.yaml:9]: Done.

Comment (ramessesii2) [reply gitops/argocd/argo-apps/pvc-cleaner.yaml:9]: > Argo CD applications are being installed in openshift-gitops ns.

Hey @AndrienkoAleksandr, I meant this `pvc-cleaner` Argo CD application to have `spec.destination.namespace:openshift-gitops`.
-> You can keep the Argo CD application destination namespace separate from the namespace where the application/operator controller itself will reside. This is also something that we're following for other operators as well like - [Openshift Pipelines](https://github.com/openshift-pipelines/pipelines-service/blob/main/gitops/argocd/tektoncd/openshift-operator.yaml#L5)
-> Please revert to the desired namespace for pvc-cleaner where `pvc-cleaner` controller will reside.  
Sorry for the confusion.

File: gitops/argocd/pvc-cleaner/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: openshift-gitops
```
Comment (ramessesii2) [inline gitops/argocd/pvc-cleaner/kustomization.yaml:4]: Please revert this to the `pvc-cleaner` ns.

**Review by ramessesii2 — CHANGES_REQUESTED** [review]

File: gitops/argocd/pvc-cleaner/kustomization.yaml (hunk 5 above)
Comment (AndrienkoAleksandr) [reply gitops/argocd/pvc-cleaner/kustomization.yaml]: Done

**Review by ramessesii2 — APPROVED** [review]
Thank you @AndrienkoAleksandr. This looks good to me. 
Please squash the commits.

**Comment by AndrienkoAleksandr** [comment]
> @ramessesii2 Please squash the commits.

Thanks, done.

@Roming22 is it ok to merge pull request ?

**Comment by Roming22** [comment]
As discussed, we'll need to have a better understanding as a team with how we need to deal with PVCs. We'll keep the PR opened until then.

**Comment by Roming22** [comment]
I've set the PR to `draft` to prevent merging for the time being.

**Comment by Roming22** [comment]
@AndrienkoAleksandr This PR has been in limbo for a while now, and has gone stale. My understanding is that we don't need that mechanism for the moment.
If you agree, would you mind closing that PR.

**Pull request closed without merging**